### DISCUSSION TITLE
There should be a config to run this fork on 8GBVram faster with un-optimzed mode.

### DISCUSSION THREAD

**Discussion by mhnoni** [basujindal/stable-diffusion#158] [Ideas]
Hi, so this fork is great for small VRam, with only 2.4 GB, with turbo it takes 3.4GB, but its still slow vs un-optimized of other forks like hlky 49sec here vs hlky fork 37sec that takes 6.4GB. So it would be cool if there is a config to enable un-optimized for people who owns > 8Gb VRam. 

I used my gtx 1070 8GB here for testing.

Thanks!

**Comment by bitRAKE** [comment]
The 1070 has faster FP32 (single-precision) floating-point performance than FP16 (half-precision) performance. By using the ```--precision full``` command line option you can take advantage of this older performance profile.

I'm in the same boat with my 1080ti. So, I understand what you are experiencing.

More modern cards have been designed with greater FP16 performance.

This wiki table also shows the legacy performance of the 10 Series cards:
https://en.wikipedia.org/wiki/GeForce_10_series#GeForce_10_(10xx)_series_for_desktops

Hard to believe it's only been five years. 😃

**Reply by mhnoni** [reply]
> --precision full

I just tried the full precision, it added 1GB Vram and made it slow from the normal 1.20min to 2.20min of this fork, same with hlky which used to be 39sec to 120sec. I think precision full is slow no matter the graphic card.

**Reply by bitRAKE** [reply]
I think you're on to something.

On my machine I get double memory use, but no speed bump from ```--precision full```.

Now to hunt down what changed - it did work at some point. IIRC, it should be about 3-4 times faster.

**Reply by bitRAKE** [reply]
What about ``` --turbo --precision full```?

That seems to be the hot ticket on my end.

I'm guessing here, but I think turbo was integrated initially and then split. So, the speed I'm remembering was with turbo. The best average speed I get is batching a couple with turbo and full. This is how I like to work on prompt crafting - seeing results quickly. (Which is almost the same as I get with large batches and half, but I need to wait a long time to see anything - run it in the background while I do other work.)

**Reply by mhnoni** [reply]
> What about ` --turbo --precision full`?
> 
> That seems to be the hot ticket on my end.
> 
> I'm guessing here, but I think turbo was integrated initially and then split. So, the speed I'm remembering was with turbo. The best average speed I get is batching a couple with turbo and full. This is how I like to work on prompt crafting - seeing results quickly. (Which is almost the same as I get with large batches and half, but I need to wait a long time to see anything - run it in the background while I do other work.)
Still, not as fast as hlky half precision. The only way to make it faster for this fork is to not split the data which it does even with turbo if I'm not mistaken.

**Reply by bitRAKE** [reply]
If hlky half precision is faster I need to check that out ...

... looks like that went the class route to code organization - nice, it consolidates things, reduces code duplication. Just a quick glance, I think they removed all the explicit ```.half()``` conversions from the samplers - which might be slowing down full precision in this repo. I'll need to play with it some and compare.

**Reply by bitRAKE** [reply]
hlky:

Wow, try ```webui.bat --precision full --no-half```. I'm getting images popping out in 15 seconds. Downside is batching two images at 512x512 is the limit, and not much range on image size either.

Okay, now for the low side: I can't get it anywhere near this repo. Even using all the options (```webui.bat --lowvram --opt-split-attention```), I can't generate larger images - it's just using too much VRAM still. I'm going to play with it some more and try to get VRAM usage down.

With this repo I can batch 14 images at 832x576 and just saturate my GFX card, or go larger with smaller batches.

**Reply by mhnoni** [reply]
> hlky:
> 
> Wow, try `webui.bat --precision full --no-half`. I'm getting images popping out in 15 seconds. Downside is batching two images at 512x512 is the limit, and not much range on image size either.
> 
So with hlky, are you getting a faster image with --precision full --no-half? what is the time you get without those additional commands?

**Reply by bitRAKE** [reply]
Without any commands it takes 21 sec. But why can't I do a batch size of 8? VRAM ranges from 3.7 to 6.8, but a batch of 2 it jumps up to 9.8GB -- something is broken there. Batch size of 3 jumps to 10.1GB. Batch of four -- forget it!

This repo, I can do a batch of 25+.

My thinking is that the default of ```autocast``` works better in hlky\webui because of removal of explicit type casting on model use. On our machines it will default to FP32, and on modern machines default to FP16. This is because the memory use largely stays the same - regardless of parameters used.

My largest use case is generating images when I'm doing something else on the machine.

**Reply by bitRAKE** [reply]
Instead of hlky's broken repo use this one:
https://github.com/AUTOMATIC1111/stable-diffusion-webui

Note only better performance, but it correctly implements multi-headed attention to split the image. So, rendering larger dimensions works well with small VRAM. Best I can figure is hlky is copying features from this repo or some such craziness.

**Reply by mhnoni** [reply]
> Instead of hlky's broken repo use this one: https://github.com/AUTOMATIC1111/stable-diffusion-webui
> 
> Note only better performance, but it correctly implements multi-headed attention to split the image. So, rendering larger dimensions works well with small VRAM. Best I can figure is hlky is copying features from this repo or some such craziness.

I have right now 4 forks on my hdd lol, performance with [AUTOMATIC1111 was the same as hlky, but I'm just not sure why precision full is slower on my side when it's faster on your side.

**Reply by bitRAKE** [reply]
Could be a slower memory bus on the 1070. I see some of the effects of transferring larger data to/from GFX card myself - precision full is actually 10x faster on the 1080ti, but because it involves larger data transfers most of that acceleration is lost (unless it stays on the GFX card)!

**Reply by mhnoni** [reply]
> Could be a slower memory bus on the 1070. I see some of the effects of transferring larger data to/from GFX card myself - precision full is actually 10x faster on the 1080ti, but because it involves larger data transfers most of that acceleration is lost (unless it stays on the GFX card)!

Maybe I'm not doing it correctly, can you tell me how you enable the precision full with basujindal fork? 
edit: nvm, you already told me in the first comment.
